src/entity/user/UserTransaction.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import type { User } from './User';
+import type { Product } from '../Product';
+import type { RequestMeta } from '../../Context';
+
+export type UserTransactionFlags = Partial<{
+  note: string;
+}>;
+
+export type UserTransactionFlagsPublic = Pick<UserTransactionFlags, 'note'>;
+
+export type UserTransactionRequest = RequestMeta;
+
+@Entity()
+export class UserTransaction {
```

Comment:
Do we actually want to store this?
Is this what you last discussed with Viktor/Denis?

Reply:
Yes

Reply:
we'll use this id for idempotency?

Reply:
yes